ldk/javascript/src/whisper/form-parser.ts

```diff
@@ -0,0 +1,33 @@
+import { Button, Components, NewWhisper, WhisperComponentType } from ".";
+import { isForm, LdkForm } from "./form";
+
+export function parse(whisper: NewWhisper): NewWhisper {
+  const ldkForms: LdkForm[] = [];
+  const outgoingWhisper: NewWhisper = {
+    ...whisper,
+    components: []
+  };
+
+  whisper.components.forEach((component: Components) => {
+    if (isForm(component)) {
+      // Lift form components up
+      component.children.forEach(formChild => outgoingWhisper.components.push(formChild));
+
+      // Store form state
+      const ldkForm = new LdkForm(component.children);
+      ldkForms.push(ldkForm);
```

Comment:
Do we need to delete form from state if whisper is closed? Wondering if this potentially could cause a leak or overlap with new form.

Reply:
Hmm, perhaps not as your state is local to the parse function. Although should we be concerned about concurrency?

Reply:
Good thought. Let me see if I can isolate this issue with a test.

Reply:
Do you mean concurrency between forms? I don't believe so, each `Form` component will get its own internal state representation via one of these `LdkForm` object instances.